src/metadata/parse.rs

```diff
@@ -157,3 +157,146 @@ pub fn parse_level(val: &str) -> String {
+            // [org, link]
+            "Mario, w3cid 12345, Nintendo, https://mario.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // [org, email, link]
+            "Mario, w3cid 12345, Nintendo, hi@mario.com, https://mario.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                email: Some("hi@mario.com".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // [org, link, email]
+            "Mario, w3cid 12345, Nintendo, https://mario.com, hi@mario.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                email: Some("hi@mario.com".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // w3cid @ index 2
+            "Mario, Nintendo, w3cid 12345, https://mario.com, hi@mario.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                email: Some("hi@mario.com".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // w3cid @ index 3
+            "Mario, Nintendo, https://mario.com, w3cid 12345, hi@mario.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                email: Some("hi@mario.com".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // w3cid @ index 4
+            "Mario, Nintendo, https://mario.com, hi@mario.com, w3cid 12345" => Ok(Editor {
+                name: "Mario".to_owned(),
+                w3c_id: Some("12345".to_owned()),
+                org: Some("Nintendo".to_owned()),
+                email: Some("hi@mario.com".to_owned()),
+                link: Some("https://mario.com".to_owned()),
+                ..Default::default()
+            }),
+            // org ends with an email
+            "Mario, Nintendo hi@nintendo.com" => Ok(Editor {
+                name: "Mario".to_owned(),
+                org: Some("Nintendo".to_owned()),
+                org_link: Some("hi@nintendo.com".to_owned()),
+                ..Default::default()
+            }),
+            // org ends with a link
+            "Mario, Nintendo https://nintendo.com" => Ok(Editor {
```

Comment:
I approve the choice of test data :-)

Maybe add one with `Mario last-name` or something?